[nr-ebpf-agent] Sync the latest updates to the nr-ebpf-agent chart

nravada opened the pull request:
<!--
Thank you for contributing to New Relic's Helm charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements:

* https://github.com/newrelic-experimental/helm-charts/blob/master/CONTRIBUTING.md#technical-requirements

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/newrelic-experimental/helm-charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the
history. This will make it easier to identify new changes. The PR will be squashed
anyways when it is merged. Thanks.

For fast feedback, please @-mention maintainers that are listed in the Chart.yaml file.

Please make sure you test your changes before you push them. Once pushed, a Github Action
will run across your changes and do some initial checks and linting. These checks run
very quickly. Please check the results. We would like these checks to pass before we
even continue reviewing your changes.
-->
#### Is this a new chart
No
#### What this PR does / why we need it:
Updates the helm chart with the latest changes made.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Special notes for your reviewer:

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[mychartname]`)

kpattaswamy reviewed (commented):
Is it possible to add the remaining changes from eBPF repo as well here to get this chart up to date?

danielstokes commented on charts/nr-ebpf-agent/values.yaml:
```diff
@@ -1,58 +1,95 @@
+# Configures the agent to send all data through the proxy specified via the otel collector.
 proxy: ""
-# -- Drop data when service names map to an IP address.
+# Drop data where to service names mapps to an IP address
```
`mapps` misspelling

nravada replied:
Resolved.

danielstokes commented on charts/nr-ebpf-agent/templates/nr-ebpf-agent-daemonset.yaml:
```diff
@@ -1,18 +1,47 @@
 metadata:
   name: nr-ebpf-agent
   labels:
+    {{- if gt .Chart.Version "0.1.14"}}
```
For this and all the other occurrences; Why are we checking this?  This would make more sense if this "feature" was planned for a later release.  Seems to me this conditional is not needed; though I could be wrong.

nravada replied:
Removed this check in the latest commit.

nravada commented:
We have added two labels, app and component, in spec.selector. The helm upgrade check is failing in the PR due to not allowing newly added labels. I have excluded the newly added labels temporarily, then all checks are passed in the PR. I have reverted the temporarily changed. i.e., included the labels again in the latest commit. So we can neglect the helm upgrade check and proceed further.

kpattaswamy commented on charts/nr-ebpf-agent/Chart.yaml:
```diff
@@ -13,7 +13,7 @@ type: application
 # to the chart and its templates, including the app version.
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 0.1.13
+version: 0.1.14
```
Will need to bump this again

nravada replied:
Done. Bumped to 0.1.15

kpattaswamy commented on charts/nr-ebpf-agent/values.yaml:
```diff
@@ -62,11 +99,11 @@ pushPeriod: "15"
     pullPolicy: IfNotPresent
     # -- The tag of the eBPF agent image to be deployed.
-    tag: 0.0.5
+    tag: nr-ebpf-agent_0.0.5
```
Will need to create new images for the agent & client, the changes you are adding for TLS/Sampling/etc. are not present in these existing image tags so any of the respective configurations from the Helm chart will not be picked up by the containers

kpattaswamy commented on charts/nr-ebpf-agent/values.yaml:
```diff
@@ -76,16 +113,27 @@ ebpfAgent:
     pullPolicy: IfNotPresent
     # -- The tag of the eBPF client image to be deployed.
-    tag: 0.0.7
+    tag: nr-ebpf-client_0.0.7
```
Ditto here for image update

kpattaswamy commented on charts/nr-ebpf-agent/values.yaml:
```diff
@@ -1,58 +1,95 @@
+      samplingLatency: ""
 
-# -- The periodicity in seconds at which the eBPF agent pushes data to the OTel collector for export to NR.
+# The periodicity in seconds at which the eBPF client pushes data to the OTel collector for export to NR.
```
should be `agent` here

kpattaswamy commented on charts/nr-ebpf-agent/values.yaml:
```diff
@@ -1,58 +1,95 @@
-# while tcp_stats tracks TCP metrics.
-stirlingSources: "socket_tracer,tcp_stats"
-# -- The protocols to enable for tracing in the socket_tracer.
+# while tcp_stats tracks TCP metrics and jvm_stats tracks JVM metrics.
```
We need to remove the `jvm_stats` connector from the comment and environment variable since we are not exporting JVM data.

nravada replied:
Removed jvm_stats ref.

kpattaswamy commented on charts/nr-ebpf-agent/values.yaml:
```diff
@@ -1,58 +1,95 @@
+# Drop data where to service names mapps to an IP address
 dropDataIpServiceNames: true
-# -- Drop data from the kube-system namespace.
+# Drop data from to the kube-system namespace
```
We will need to prefix all of the relevant comments with `--`. This is because the ReadMe's auto generation relies on that prefix in order to add the description. 

The auto generation of the ReadMe is done by running [helm-docs](https://github.com/norwoodj/helm-docs).

kpattaswamy commented on charts/nr-ebpf-agent/values.yaml:
```diff
@@ -1,58 +1,95 @@
+# Drop data where to service names mapps to an IP address
 dropDataIpServiceNames: true
-# -- Drop data from the kube-system namespace.
+# Drop data from to the kube-system namespace
```
```suggestion
# -- Drop data from the kube-system namespace.
```

nravada replied:
Done.

nravada commented on charts/nr-ebpf-agent/values.yaml:
```diff
@@ -131,3 +131,33 @@ nodeSelector: {}
 tolerations: []
 # -- Sets all pods' affinities. Can be configured also with `global.affinity`
 affinity: {}
+# --  Kubernetes cluster domain
```
Ran helm-docs and updated the README file

mbolla22 commented on charts/nr-ebpf-agent/values.yaml:
```diff
@@ -76,16 +113,27 @@ ebpfAgent:
+  podSecurityContext: {}
+  # -- Sets ebpfAgent pod containerSecurityContext. Overrides `containerSecurityContext` and `global.securityContext.container`
+  containerSecurityContext:
+    privileged: true
```
We can remove this privileged: true as we are not exposed this to customer

dbudziwojskiNR commented on charts/nr-ebpf-agent/README.md:
````diff
@@ -40,7 +40,7 @@ kubectl logs <ebpf-pod-name> -c nr-ebpf-client -n newrelic
 ```
 
-### Check the logs of the OpenTelemetry collector pod:
+### Check the logs of the OpenteTemetry collector pod:
````
Looks like you have type in the template README

dbudziwojskiNR commented on charts/nr-ebpf-agent/templates/nr-ebpf-agent-daemonset.yaml:
```diff
@@ -1,18 +1,41 @@
 metadata:
   name: nr-ebpf-agent
   labels:
+    app: nr-ebpf-agent
```
You may not need to add an `app` label as component is typically enough

Philip-R-Beckwith replied:
Not needed, but don't think it's an issue to have well labeled resources.
